### PR TITLE
Adds example to squash commit documentation

### PR DESCRIPTION
**Pull request by rotodd**
## Description
Adds an example of a squash commit header and description.

Note: this is simply my interpretation of our current documentation! I am very open to changing this. I am mainly looking to have an example here to add some clarity.

## FHIR Team Checklist
- ✅ **Update the title** of the PR to be succinct and less than 50 characters
- ✅ **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- ✅ Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- ✅ Tag the PR with **Azure API for FHIR** if this will release to the managed service
- ✅ Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: SquashMergeRequirements.md [hunk 1]
```diff
@@ -7,10 +7,24 @@ When commiting your PR, please make sure to do these steps at Squash/Merge to as
...
+
+PR title + PR number (maximum 50 characters)
+
+`Adds CMK option to deployment templates. (#1234)`
+
+Squash/Merge description:
+
+User friendly description + GitHub or VSTS item number (maximum 72 characters)
+
+`Enables customer managed keys for Cosmos DB backed services. (AB#12345)`
```
Comment (LTA-Thinking) [inline SquashMergeRequirements.md:30]: How does the semver command fit into the 72 character limit? 
I think having one of those commands in the example would be helpful.

**Comment by LTA-Thinking** [comment]
Do we want a uniform title/description for the dependabot PRs? The names it generates for PRs are often too long for the heading.

**Review by LTA-Thinking — APPROVED** [review]

File: SquashMergeRequirements.md [hunk 2]
```diff
@@ -7,10 +7,24 @@ When commiting your PR, please make sure to do these steps at Squash/Merge to as
...
 
 1. **Create a new issue** in GitHub that describes the warning/issue
 1. Tag the PR with **KI-Breaking** or **KI-Warning**
 1. **Reference the new GitHub issue** from the PR description
+
+## Example
+
+Squash/Merge commit heading:
+
+PR title + PR number (maximum 50 characters)
```
Comment (namadabu) [inline SquashMergeRequirements.md:22]: Just for my clarification, is the PR number also included in the 50 character limit?

File: SquashMergeRequirements.md (hunk 1 above)
Comment (brendankowitz) [reply SquashMergeRequirements.md:30]: 72 chars is the recommending line-wrap limit, you can have multiple lines, for example if you wanted to do:

```
Enables customer managed keys for Cosmos DB backed services. 
Refs AB#12345
+semver: patch
```

That would be ok

Comment (rotodd) [reply SquashMergeRequirements.md:30]: Ah, that makes sense. Thanks! I will add this to the doc.

File: SquashMergeRequirements.md (hunk 2 above)
Comment (rotodd) [reply SquashMergeRequirements.md:22]: Good question. I would assume the PR number is included but I am not 100% sure. @brendankowitz would you happen to know how this affects the release notes? I can also defer to @CaitlinV39 when she is back on Monday.

File: SquashMergeRequirements.md (hunk 1 above)
Comment (rotodd) [reply SquashMergeRequirements.md]: Updated in 0074627.

**Comment by rotodd** [comment]
> Do we want a uniform title/description for the dependabot PRs? The names it generates for PRs are often too long for the heading.

@LTA-Thinking Good question, I'm not sure. A relevant note that might be helpful: I asked @CaitlinV39 about #1340 when I was merging it, wondering if I should shorten auto-generated name, and Caitlin said it was alright to leave it for now. I think we might have plans to shorten it, but I will defer to her to answer fully.

File: SquashMergeRequirements.md (hunk 2 above)
Comment (brendankowitz) [reply SquashMergeRequirements.md:22]: I think all characters are included. Lets just see how we can and assess how challenging this is :)

Comment (rotodd) [reply SquashMergeRequirements.md]: Clarified this in 0074627.

**Review by CaitlinV39 — APPROVED** [review]
Thanks so much for doing this!

**Comment by rotodd** [comment]
> > Do we want a uniform title/description for the dependabot PRs? The names it generates for PRs are often too long for the heading.
> 
> @LTA-Thinking Good question, I'm not sure. A relevant note that might be helpful: I asked @CaitlinV39 about #1340 when I was merging it, wondering if I should shorten auto-generated name, and Caitlin said it was alright to leave it for now. I think we might have plans to shorten it, but I will defer to her to answer fully.

@LTA-Thinking chatted with @CaitlinV39 and confirmed that it is fine to leave auto-generated PRs as-is for now. I specified this in the docs in 0074627.

**Pull request merged**